src/Octoshift/GithubApi.cs

```diff
@@ -459,5 +460,145 @@ public virtual async Task<string> GetArchiveMigrationUrl(string org, int migrati
             var response = await _client.GetNonSuccessAsync(url, HttpStatusCode.Found);
             return response;
         }
+
+        public virtual async Task<Mannequin> GetMannequin(string orgid, string username)
+        {
+            var url = $"{_apiUrl}/graphql";
+
+            var payload = BuildListMannequinPayload(orgid, null);
+
+            JObject data;
+            do
+            {
+                var response = await _client.PostAsync(url, payload);
+                data = JObject.Parse(response);
+
+                var mannequin = data["data"]["node"]["mannequins"]["nodes"].FirstOrDefault(node =>
+                    { return username.Equals((string)node["login"], System.StringComparison.OrdinalIgnoreCase); }
+                );
+
+                if (mannequin != null)
+                {
+                    Claimant claimant = null;
+
+                    if (mannequin["claimant"].Any())
+                    {
+                        claimant = new Claimant
+                        {
+                            Id = (string)mannequin["claimant"]["id"],
+                            Login = (string)mannequin["claimant"]["login"]
+                        };
+                    }
+
+                    return new Mannequin
+                    {
+                        Id = (string)mannequin["id"],
+                        Login = (string)mannequin["login"],
+                        MappedUser = claimant
+
+                    };
+                }
+
+                payload = BuildListMannequinPayload(orgid, (string)data["data"]["node"]["mannequins"]["pageInfo"]["endCursor"]);
+
+            } while ((bool)data["data"]["node"]["mannequins"]["pageInfo"]["hasNextPage"]);
+
+            return new Mannequin();
+        }
+
+        public virtual async Task<string> GetUserId(string login)
+        {
+            var url = $"{_apiUrl}/graphql";
+
+            var payload = new
+            {
+                query = "query($login: String!) {user(login: $login) { id, name } }",
+                variables = new { login }
+            };
+
+            var response = await _client.PostAsync(url, payload);
+            var data = JObject.Parse(response);
+
+            return data["data"]["user"].Any() ? (string)data["data"]["user"]["id"] : null;
```

Comment:
Is the data["data"]["user"] going to be `null` here?

Reply:
yes, in case the user entered a nonexistant user